Framework/Algorithms/src/DiffractionFocussing2.cpp

```diff
@@ -211,13 +212,16 @@ void DiffractionFocussing2::exec() {
       // This is the input spectrum
       const auto &inSpec = m_matrixInputW->getSpectrum(inWorkspaceIndex);
       // Get reference to its old X,Y,and E.
-      const MantidVec &Xin = inSpec.readX();
-      const MantidVec &Yin = inSpec.readY();
-      const MantidVec &Ein = inSpec.readE();
+      auto &Xin = inSpec.x();
+      auto &Yin = inSpec.y();
+      auto &Ein = inSpec.e();
 
       outSpec.addDetectorIDs(inSpec.getDetectorIDs());
       try {
-        VectorHelper::rebinHistogram(Xin, Yin, Ein, Xout, Yout, Eout, true);
```

Comment:
DiffractionFocussing is partially updated because the use of `VectorHelper` methods restricts the use of the `HistogramData` types. This `VectorHelper::rebinHistogram` method should ideally be implemented in histogram.